pkg/autodiscovery/configresolver/configresolver.go

```diff
@@ -463,15 +464,22 @@ func tagsAdder(tags []string) func(interface{}) error {
 					}
 				}
 			}
+
 			for _, t := range tags {
 				tagSet[t] = struct{}{}
 			}
-			typedTree["tags"] = make([]string, len(tagSet))
+
+			allTags := make([]string, len(tagSet))
+
 			i := 0
 			for k := range tagSet {
-				typedTree["tags"].([]string)[i] = k
+				allTags[i] = k
 				i++
 			}
+
+			sort.Strings(allTags)
+
+			typedTree["tags"] = allTags
```

Comment:
Should we consider the InsertSort available in `pkg/util/sort.go` instead? I think this is run often during the Agent runtime and could help a bit reducing the GC pressure.

Reply:
Discussed on slack, but repeating it here:

* This merges static tags with the ones from the tagger. That's very often larger than `InsertionSortThreshold`. Or put it another way: on environments where it's larger, it's almost always larger.
* This runs only for file configs, only once per file, and afaict only at agent start.

Will keep it in mind though if there's performance degradation.